compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/res/Resources.desktop.kt

```diff
@@ -35,15 +55,68 @@ inline fun <T> useResource(
 /**
  * Open [InputStream] from a resource stored in resources for the application.
  *
+ * @param resourcePath  path of resource in loader
+ * @param loader  resource loader
+ *
  * @throws IllegalArgumentException if there is no [resourcePath] in resources
  */
 @PublishedApi
-internal fun openResource(resourcePath: String): InputStream {
-    // TODO(https://github.com/JetBrains/compose-jb/issues/618): probably we shouldn't use
-    //  contextClassLoader here, as it is not defined in threads created by non-JVM
-    val classLoader = Thread.currentThread().contextClassLoader!!
+internal fun openResource(
+    resourcePath: String,
+    loader: ResourceLoader
+): InputStream {
+    return loader.load(resourcePath)
+}
 
-    return requireNotNull(classLoader.getResourceAsStream(resourcePath)) {
-        "Resource $resourcePath not found"
+/**
+ * Open [InputStream] from a resource stored in resources for the application.
+ *
+ * @param resourcePath  path of resource
+ *
+ * @throws IllegalArgumentException if there is no [resourcePath] in resources
+ */
+@PublishedApi
+internal fun openResource(
+    resourcePath: String,
+): InputStream {
+    return defaultResourceLoader().load(resourcePath)
+}
+
+/**
+ * Abstraction for loading resources. This API is intended for use in synchronous cases,
+ * where resource is expected to be loaded quick during the first composition, and so potentially
+ * slow operations like network access is not recommended. For such scenarious use functions
+ * [loadSvgPainter] and [loadXmlImageVector] instead on IO dispatcher.
+ */
+interface ResourceLoader {
+    fun load(resourcePath: String): InputStream
+}
+
+/**
+ * Resource loader based on JVM current context class loader.
+ */
+class ClassLoaderResourceLoader : ResourceLoader {
+    override fun load(resourcePath: String): InputStream {
+        // TODO(https://github.com/JetBrains/compose-jb/issues/618): probably we shouldn't use
+        //  contextClassLoader here, as it is not defined in threads created by non-JVM
+        return Thread.currentThread().contextClassLoader!!.getResourceAsStream(resourcePath)
+    }
+}
+
+/**
+ * Resource loader from the file system relative to a certain root location.
+ */
+class FileResourceLoader(val root: File) : ResourceLoader {
+    override fun load(resourcePath: String): InputStream {
+        return FileInputStream(File(root, resourcePath))
     }
-}
\ No newline at end of file
+}
+
+/**
+ * Resource loader which is capable to load resources from `resources` folder in an application's
+ * project. Ability to load from dependent modules resources is not guaranteed in the future.
+ * Use explicit `ClassLoaderResourceLoader` instance if such guarantee is needed.
+ */
+fun defaultResourceLoader(): ResourceLoader = _defaultResourceLoader
```

Comment:
In Compose style it would be:
```
interface ResourceLoader {
    companion object {
       val Default = ClassLoaderResourceLoader()
    }
}
```

Reply:
Ok